### PR TITLE
fix: make it possible to add multiple mountingOverrideDelegates

### PR DESCRIPTION
**Pull request by WoLewicki**
## Summary:

PR changing the single mountingOverrideDelegate to a vector of those, so other listeners can operate on the transaction. Used by `react-native-screens` in https://github.com/software-mansion/react-native-screens/pull/2134 and `react-native-reanimated` in https://github.com/software-mansion/react-native-reanimated/pull/6055.

Till now, only one listener could be added there, meaning that e.g. `Layout Animations` from `react-native`,  `Layout Animations` from `react-native-reanimated` and listening for `Screen` removal in `react-native-screens` could not operate at the same time.

## Changelog:

[GENERAL] [FIXED] - Add option for multiple `mountingOverrideDelegates`

## Test Plan:

The code of `LayoutAnimations` inside `react-native` should work the same since it will add just one listener then. For other cases, different libraries can read/mutate transactions.

### CODE REVIEW

**Comment by WoLewicki** [comment]
I am aware that the name of the method should be probably changed from `setMountingOverrideDelegate` to `addMountingOverrideDelegate` but this way we keep the compatibility between versions.

File: packages/react-native/ReactCommon/react/renderer/mounting/MountingCoordinator.cpp [hunk 1]
```diff
@@ -104,31 +104,35 @@ std::optional<MountingTransaction> MountingCoordinator::pullTransaction()
...
-      number_++;
-      telemetry.willLayout();
-      telemetry.didLayout();
-      telemetry.willCommit();
-      telemetry.didCommit();
-      telemetry.willDiff();
-      telemetry.didDiff();
+  for(auto it = mountingOverrideDelegates_.begin(); it < mountingOverrideDelegates_.end(); it++)
+  {
+    auto &delegate = *it;
```
Comment (j-piasecki) [inline packages/react-native/ReactCommon/react/renderer/mounting/MountingCoordinator.cpp:109]:
```suggestion
  for (const auto &delegate : mountingOverrideDelegates_)
  {
```

Not sure about const

File: packages/react-native/ReactCommon/react/renderer/mounting/MountingCoordinator.cpp [hunk 2]
```diff
@@ -203,7 +206,7 @@ ShadowTreeRevision MountingCoordinator::getBaseRevision() const {
 void MountingCoordinator::setMountingOverrideDelegate(
```
Comment (sammy-SC) [inline packages/react-native/ReactCommon/react/renderer/mounting/MountingCoordinator.cpp:206]: shouldn't this be renamed to 'addMountingOverrideDelegate'?

**Comment by sammy-SC** [comment]
looks good. I'm importing the diff. If you manage to do the rename, that's great. Otherwise, it can be done in subsequent diff. 

Thank you

**Comment by WoLewicki** [comment]
As mentioned in https://github.com/facebook/react-native/pull/44927#issuecomment-2165858320, not changing the name would make it work for different RN versions, but since we don't care too much about it on new arch right now, I'll change it 😅

**Comment by WoLewicki** [comment]
@sammy-SC changed from `set` to `add`.

**Pull request closed without merging**